app/assets/javascripts/purchaseFlow/PurchaseFlowWrapper.tsx

```diff
@@ -27,8 +29,11 @@ export const loadPurchaseFlowUrl = async (
   application: WebApplication
 ): Promise<boolean> => {
   const url = await getPurchaseFlowUrl(application);
+  const params = new URLSearchParams(window.location.search);
+  const period = params.get('period') ?? '';
+  const plan = params.get('plan') ?? '';
   if (url) {
-    window.location.assign(url);
+    window.location.assign(`${url}&period=${period}&plan=${plan}`);
```

Comment:
Feel uneasy about a url that could look like `.com/purchase?token=123&period=&plan=`. We should instead conditionally build the string based on availability of params.